src/Symfony/Component/DependencyInjection/Argument/RewindableGenerator.php

```diff
@@ -14,13 +14,21 @@
 /**
  * @internal
  */
-class RewindableGenerator implements \IteratorAggregate
+class RewindableGenerator implements \IteratorAggregate, \Countable
 {
     private $generator;
+    private $count;
 
-    public function __construct(callable $generator)
+    /**
+     * RewindableGenerator constructor.
+     *
+     * @param callable $generator
+     * @param int|null $count If provided, used as the returned value for count. iterator_to_array will be used otherwise.
+     */
+    public function __construct(callable $generator, $count = null)
```

Comment:
As this class is internal and used only for this purpose, I think I can drop the `$count = null` behavior, though.